### PR TITLE
Fix undefined index precision

### PR DESCRIPTION
**Pull request by borjagarciab**
Fix this error
Argument 1 passed to PrestaShop\PrestaShop\Core\Localization\CLDR\ComputingPrecision::getPrecision() must be of the type int, null given, called in /var/www/vhost/xxx.com/home/html/classes/Context.php on line 489

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | When you use this method with a currency without this variable, php throw this exception: <br>Argument 1 passed to PrestaShop\PrestaShop\Core\Localization\CLDR\ComputingPrecision::getPrecision() must be of the type int, null given, called in /var/www/vhost/xxx.com/home/html/classes/Context.php on line 489
| Type?             | bug fix 
| Category?         |  FO
| BC breaks?        | yes 
| Deprecations?     | no
| Fixed ticket?     | -
| Related PRs       |  https://github.com/PrestaShop/PrestaShop/pull/30167
| How to test?      | Use a currency with no currency on context
| Possible impacts? | Do not crash when no currency->precision

### CODE REVIEW

**Comment by borjagarciab** [comment]
Hi,
Sometime ago , i request a similar PR https://github.com/PrestaShop/PrestaShop/pull/30167

Today i was developing a module and i get same error.

In my module , i create customizations for a product when customer click on add to cart. (Maybe is not the best method)
image

On the add to cart process, the module ps_googleanalitics crash on (ps_googleanalytics/classes/Hook/HookActionCartSave) line 70.
This crash give me a 500 error on add to cart.

Debugging , i finally get the precision error , because the currency is not set.
The best solution i found is set the main currency.


Thanks.

**Review by matks — DISMISSED** [review]
Well this change makes sense for me. The real problem is that `Context` class is flawed but I think the code you add makes it easier to use 👍 

Thank you!

**Comment by borjagarciab** [comment]
Hi!
Should i do anything?
Thanks!

**Comment by SharakPL** [comment]
Please fix coding standards as described [here](https://devdocs.prestashop-project.org/8/modules/testing/basic-checks/#coding-standards) or check the tests where it's red and manually add suggested changes.

This command from project root should fix all issues:
```sh
php vendor/bin/php-cs-fixer fix classes/Context.php
```

**Comment by kpodemski** [comment]
Hello @borjagarciab 

There are two problems with your PR:
- integration tests fail
- coding standards issues

File: classes/Context.php [hunk 1]
```diff
@@ -554,9 +554,13 @@ public function getComputingPrecision()
     {
         if ($this->priceComputingPrecision === null) {
             $computingPrecision = new ComputingPrecision();
+            //If currency is not set, we use the default currency
+            if(!isset($this->currency)){
+                $currency = new Currency(Configuration::get('PS_CURRENCY_DEFAULT'));
```
Comment (kpodemski) [inline classes/Context.php]:
```suggestion
            // If the currency is not set, we use the default currency
            if (!isset($this->currency)) {
                $currency = new Currency((int) Configuration::get('PS_CURRENCY_DEFAULT'));
```

**Comment by matks** [comment]
Hi,

Since we had no news from you for more than 2 months, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**